CMakeLists.txt

```diff
@@ -1,8 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
-
-if(POLICY CMP0077)
-    cmake_policy(SET CMP0077 NEW)
-endif()
+cmake_minimum_required(VERSION 3.1...3.17.2)
```

Comment:
The ``3.1`` change makes sense. Everything starting from ``...`` is not required. I also havn't ever seen a CMake file specifying a "max version".

About the CMP0077 removal: This is wrong.

Reply:
No, it's not wrong. See https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html#policy-settings  
The minimum version is for erroring out if the CMake version is too old, but the maximum version is for policies. `CMP0077` was introduced in `3.13` so a max of `3.17` will set it to `NEW`.

If you somehow don't trust the CMake documentation, then it's easy to verify these:
```cmake
cmake_minimum_required(VERSION 3.1...3.17)
cmake_policy(GET CMP0077 value)
message(NOTICE "${value}")
```
`cmake -P test.cmake` will print `NEW`.